ui/app/common/displaycontrols/programs/directives/programs.js

```diff
@@ -55,6 +60,19 @@ angular.module('bahmni.common.displaycontrol.programs')
                 var isCodedConceptFormat = function (format) {
                     return format == "org.bahmni.module.bahmnicore.customdatatype.datatype.CodedConceptDatatype";
                 };
+                $scope.translateAttributeName = function (attribute, moduleName) {
+                    var keyPrefix = moduleName && modulePrefixMap[moduleName] ? modulePrefixMap[moduleName] : '';
+                    if (typeof attribute.description == 'undefined') {
+                        attribute.description = attribute.display;
+                    }
+                    var keyName = attribute.description.toUpperCase().replace(/[^a-zA-Z0-9]/g, "").replace(/ /g, "_").replace(/-/g, "_");
+                    var translationKey = keyPrefix + '_' + keyName;
+                    var translation = $translate.instant(translationKey);
+                    if (translation == translationKey) {
+                        return attribute.description;
+                    }
+                    return translation;
+                };

```

Comment:
@buvaneswari-arun I see the similar method has been used in multiple places. Can we move this to a util file

Reply:
Done. TranslationUtil.js is newly added and this method is moved there.